apps/_demo/index.js

```diff
@@ -0,0 +1,120 @@
+const express = require('express')
+const { Keystone } = require('@keystonejs/keystone')
+const { PasswordAuthStrategy } = require('@keystonejs/auth-password')
+const { GraphQLApp } = require('@keystonejs/app-graphql')
+const { AdminUIApp } = require('@keystonejs/app-admin-ui')
+const { StaticApp } = require('@keystonejs/app-static')
+const { NextApp } = require('@keystonejs/app-next')
+const { createItems } = require('@keystonejs/server-side-graphql-client')
+
+const conf = require('@core/config')
+const access = require('@core/keystone/access')
+const { areWeRunningTests } = require('@core/keystone/test.utils')
+const { EmptyApp } = require('@core/keystone/test.utils')
+const { prepareDefaultKeystoneConfig } = require('@core/keystone/setup.utils')
+const { registerSchemas } = require('@core/keystone/schema')
+
+const keystone = new Keystone({
+    ...prepareDefaultKeystoneConfig(conf),
+    onConnect: async () => {
+        // Initialise some data
+        if (conf.NODE_ENV !== 'development') return // Just for dev env purposes!
+        // This function can be called before tables are created! (we just ignore this)
+        try {
+            const users = await keystone.lists.User.adapter.findAll()
+            if (!users.length) {
+                const initialData = require('./initialData')
+                for (let { listKey, items } of initialData) {
+                    console.log(`🗿 createItems(${listKey}) -> ${items.length}`)
+                    await createItems({
+                        keystone,
+                        listKey,
+                        items,
+                    })
+                }
+            }
+        } catch (e) {
+            console.warn('onConnectError:', e)
+        }
+    },
+})
+
+registerSchemas(keystone, [
+    require('./schema/User'),
+    require('./schema/Organization'),
+    require('./schema/Auth'),
+    require('./schema/Test'),
+])
+
+class CustomAppExample {
+    prepareMiddleware ({ keystone, dev, distDir }) {
+        const app = express()
+        app.use(function (req, res, next) {
+            req._CustomApp_requestLevelData = { foo: 22 }
+            // console.log('Time:', Date.now())
+            next()
+        })
+        return app
+    }
+}
+
+class CustomGQLPluginExample {
+    serverWillStart () {
+        console.log('Server starting!')
+    }
+
+    requestDidStart (requestContext) {
+        if (conf.NODE_ENV !== 'production') return
+        // console.log('GQL!o :', requestContext.operation)
+        console.log('GQL!op:', requestContext.request.operationName)
+        console.log('GQL!q :', requestContext.request.query.replace(/\n/g, '').replace(/\s+/g, ' '))
+        // console.log('GQL!v :', requestContext.request.variables)
```

Comment:
Too many console.logs and comments, should they remain here?

Reply:
It's OK for demo staff, not for examples ;) 
I can delete it :)